velox/expression/tests/ExpressionVerifier.cpp

```diff
@@ -68,11 +68,33 @@ bool defaultNullRowsSkipped(const exec::ExprSet& exprSet) {
   }
   return false;
 }
+
+// Returns a RowVector with only the rows specified in 'rows'. Does not maintain
+// encodings.
+RowVectorPtr reduceToSelectedRows(
+    const RowVectorPtr& rowVector,
+    const SelectivityVector& rows) {
+  if (rows.isAllSelected()) {
+    return rowVector;
+  }
+  BufferPtr indices = allocateIndices(rows.end(), rowVector->pool());
+  auto rawIndices = indices->asMutable<vector_size_t>();
+  vector_size_t cnt = 0;
+  rows.applyToSelected([&](vector_size_t row) { rawIndices[cnt++] = row; });
+  VELOX_CHECK_GT(cnt, 0);
+  indices->setSize(cnt * sizeof(vector_size_t));
+  RowVectorPtr reducedVector = std::dynamic_pointer_cast<RowVector>(
+      BaseVector::create(rowVector->type(), cnt, rowVector->pool()));
+  SelectivityVector rowsToCopy(cnt);
+  reducedVector->copy(rowVector.get(), rowsToCopy, rawIndices);
```

Comment:
nit: This piece of code can be simply replaced with `BaseVector::wrapInDictionary(nullptr, indices, cnt, rowVector)`?

Reply:
True, the only reason I did this was to avoid creating a top level RowVector which is dictionary encoded which we do not expect in velox. This basically reduces some steps if we did (wrapInDictionary + flattenVector) instead.

This however will only be used by the presto runner, where it will probably write this to disk first. I have not had a chance to run it with PQR yet, but if that codepath supports an encoded top level row vector then we should be good and I can replace this with your suggestion.